### PR TITLE
Add heart textures for chests, pots and crates

### PR DESCRIPTION
**Pull request by AngheloAlf**
Add a specific texture for the "Texture (Match content)" option for chests, pots and crates.
Code for beehives was also added but commented out like the rest of the beehive code.

This heart texture will be used if the item contains either a "Piece of Heart", "Heart Container" ~~or "Double Defense"~~.

I had to add a few changes to allow changing the top textures of the pots because it looked weird with the default texture instead of a custom one

## Chest
image

## Pot
image

## Crate
image
image

## Small crate
image

Thanks a lot to `C♭#6719` on discord for providing this awesome heart textures!

### CODE REVIEW

**Comment by ETR-BTF** [comment]
I really like the pot textures! Great work!

File: ASM/c/item_table.h [hunk 1]
```diff
@@ -22,6 +22,7 @@ typedef enum {
     /* 13 */ SILVER_CHEST,
     /* 14 */ SKULL_CHEST_SMALL,
     /* 15 */ SKULL_CHEST_BIG,
+    /* 16 */ HEART_CHEST,
```
Comment (fenhl) [inline ASM/c/item_table.h]: There should be small and big versions of the heart chest type that work like `SKULL_CHEST_SMALL` and `SKULL_CHEST_BIG`.

Comment (AngheloAlf) [reply ASM/c/item_table.h]: I didn't make small and big versions because most chests don't have that variation, but I guess I could do that

**Comment by fenhl** [comment]
I'm not sure if double defence should use this new texture. Token chest textures were introduced to make token wincons play better, and this would be the equivalent of that for heart wincons, which don't include double defence.

File: ASM/c/item_table.h (hunk 1 above)
Comment (fenhl) [reply ASM/c/item_table.h]: If you search the code for `SKULL_CHEST_SMALL` and `SKULL_CHEST_BIG` you'll find some logic for making token chests big if there's a token win condition. There's similar code that makes heart pieces and containers use `GILDED_CHEST` if there's a heart wincon. That code should be changed to use a new `HEART_CHEST_BIG` type.

Comment (AngheloAlf) [reply ASM/c/item_table.h]: This change is a bit weird because `HEART_CHEST_BIG` goes unused. :man_shrugging:

Comment (fenhl) [reply ASM/c/item_table.h]: It shouldn't go unused. I'll send a quick PR to show what I mean.

Comment (fenhl) [reply ASM/c/item_table.h]: AngheloAlf#1

Comment (AngheloAlf) [reply ASM/c/item_table.h]: Merged, thanks!!

**Pull request merged**